### PR TITLE
TRUNK-5830 : Added hibernate annotations for FieldType

### PR DESCRIPTION
**Pull request by LuGO0**
## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-5830

## Checklist: I completed these to help reviewers :)

- [ ] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [ ] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [ ] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [ ] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by LuGO0** [comment]
please review @gcliff @dkayiwa

**Review by tendomart — APPROVED** [review]
Look good to me..

let's wait for the build to finish

**Comment by LuGO0** [comment]
@tendomart fixed the build errors, but this error comes up
```org.hibernate.tool.schema.spi.SchemaManagementException: Schema-validation: missing column [date_changed] in table [field_type]
```
but this column is part of the base class :(

File: api/src/main/java/org/openmrs/FieldType.java [hunk 1]
```diff
@@ -9,17 +9,35 @@
...
 	public static final long serialVersionUID = 35467L;
 	
 	// Fields
-	
+	@Id
+	@Column(name = "field_type_id")
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
 	private Integer fieldTypeId;
-	
+
```
Comment (HerbertYiga) [inline api/src/main/java/org/openmrs/FieldType.java:39]: lets remove this space

**Comment by dkayiwa** [comment]
> fixed the build errors

Is this correct? The pull request does not seem to reflect so.

File: api/src/main/java/org/openmrs/FieldType.java [hunk 2]
```diff
@@ -9,17 +9,35 @@
...
+		@AttributeOverride(name = "retired", column = @Column(name = "retired", columnDefinition = "boolean default false"))
+})
 public class FieldType extends BaseChangeableOpenmrsMetadata {
 	
 	public static final long serialVersionUID = 35467L;
 	
 	// Fields
-	
+	@Id
+	@Column(name = "field_type_id")
```
Comment (HerbertYiga) [inline api/src/main/java/org/openmrs/FieldType.java:36]: do all your attributes correspond with what he have in the hibernate mapping file that was removed?

**Comment by LuGO0** [comment]
> > fixed the build errors
> 
> Is this correct? The pull request does not seem to reflect so.

@dkayiwa I fixed the older build errors then this error came up.

> > fixed the build errors
> 
> Is this correct? The pull request does not seem to reflect so.

No @dkayiwa fixed the older build errors caused by missing attribute overrides for retired attribute

File: api/src/main/java/org/openmrs/FieldType.java (hunk 2 above)
Comment (LuGO0) [reply api/src/main/java/org/openmrs/FieldType.java:36]: No @HerbertYiga dosent seem so from the build errors.

**Comment by tendomart** [comment]
@LuGO0 hope you're still looking into the failures .

**Comment by tendomart** [comment]
When you look at the table , does that column exist ? i.e **date_changed**

**Comment by LuGO0** [comment]
> When you look at the table , does that column exist ? i.e `date_changed`

where can I look for the table @tendomart ? workBench?

**Comment by tendomart** [comment]
Yes you can use workbench or through terminal.

**Comment by LuGO0** [comment]
testing DB
second

^^ only these columns are there when I inspect using cmd prompt. these are the same columns that were present in the field_type.hbm.xml file.



I had one more question, are these updated when we build and run openmrs core using the openmrs-sdk?

**Comment by LuGO0** [comment]
Closing this PR for now, will reopen if I fix the Issue.

**Pull request closed without merging**